streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala

```diff
@@ -123,6 +126,12 @@ class JobGenerator(jobScheduler: JobScheduler) extends Logging {
         timedOut
       }
 
+      // generate one more bacth to make sure RDD in lastJob is checkpointed.
+      if (!jobScheduler.receiverTracker.hasUnallocatedBlocks &&
+        ssc.graph.isCheckpointMissedLastTime) {
```

Comment:
Can you add more comment to explain why we need to check the `ssc.graph.isCheckpointMissedLastTime`?